gst/nnstreamer/tensor_converter/tensor_converter.c

```diff
@@ -691,6 +710,51 @@ _gst_tensor_converter_chain_timestamp (GstTensorConverter * self,
   }
 }
 
+/** @brief Chain function's private routine to push buffer into src pad */
+static GstFlowReturn
+_gst_tensor_converter_chain_push (GstTensorConverter * self, GstBuffer * buf)
+{
+  GstBuffer *buffer = buf;
+
+  if (self->in_media_type == _NNS_OCTET) {
+    /* configure multi tensors */
+    if (self->tensors_info.num_tensors > 1) {
+      GstMemory *mem;
+      GstMapInfo info;
+      gpointer data;
+      gsize idx, size;
+      guint i;
+
+      g_assert (self->frames_per_tensor == 1);
+
+      buffer = gst_buffer_new ();
+
+      mem = gst_buffer_get_all_memory (buf);
+      gst_memory_map (mem, &info, GST_MAP_READ);
+
+      idx = 0;
+      for (i = 0; i < self->tensors_info.num_tensors; ++i) {
+        size = gst_tensors_info_get_size (&self->tensors_info, i);
+        data = g_memdup (info.data + idx, size);
+        idx += size;
+
+        gst_buffer_append_memory (buffer,
+            gst_memory_new_wrapped (0, data, size, 0, size, data, g_free));
+      }
+
+      gst_memory_unmap (mem, &info);
+      gst_memory_unref (mem);
+
+      /* copy timestamps */
+      gst_buffer_copy_into (buffer, buf, GST_BUFFER_COPY_METADATA, 0, -1);
```

Comment:
Q. why does it have to separate data/metadata copies (i.e., not using `GST_BUFFER_COPY_MEMORY`) ?

Reply:
The input buffer from byte stream has one memory block. If user configures multiple tensors as output buffer, tensor-converter should divide one memory into multiple blocks (multi tensors).

Reply:
Oh, thanks for the explanation.